src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -2082,35 +2049,24 @@ private static bool IsMetachar(char ch)
          */
         private void AddConcatenate(int pos, int cch, bool isReplacement)
         {
-            RegexNode node;
-
             if (cch == 0)
                 return;
 
+            RegexNode node;
             if (cch > 1)
             {
                 string str;
                 if (UseOptionI() && !isReplacement)
                 {
                     str = string.Create(cch, (_pattern, _culture, pos, cch), (span, state) =>
-                    {
-                        ReadOnlySpan<char> input = state._pattern.AsSpan(pos, cch);
-
-                        // We do the ToLower character by character for consistency.  With surrogate chars, doing
```

Comment:
Same concern as in the other place

Reply:
> Yes we have several tests that hit these code paths. Our test bed is fortunately very rich.

I do not see any tests in the test bed that exercise surrogates that this comment is talking about.

Reply:
Because our engine operates on UTF16 code units and does not support surrogate pairs?

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/character-classes-in-regular-expressions#SupportedNamedBlocks

Reply:
We should still have tests for it, even if it is documented as unsupported in the documentation. We should not be crashing 

The comment you are removing suggests that handling of surrogates showed up as a problem in the past and that somebody tried to make it better/more predictable at least.